checkstyle-tester/diff.groovy

```diff
@@ -153,6 +156,21 @@ def isExistingGitBranch(gitRepo, branchName) {
     return exist
 }
 
+def copyConfigFilesAndUpdatePaths(configFilesList) {
+    // Remove boolean value for single config in case of different base and patch config
+    configFilesList.removeIf { it instanceof Boolean };
+    // Remove dups in case of single config
+    configFilesList = configFilesList.unique();
+
+    for (filename in configFilesList) {
+        def sourceFile = new File(filename)
+        def checkstyleTesterDir = new File("").getCanonicalFile()
+        def destFile = new File("${checkstyleTesterDir.toPath()}/${sourceFile.getName()}")
+        Files.copy(sourceFile.toPath(), destFile.toPath(), REPLACE_EXISTING)
```

Comment:
Why do the files have to be copied? Why can't they be used from where they are?

This will cause extra files to appear in working directory, that will not be cleared away and show up as unstaged changes.

Reply:
Due to the fact that config files are created in different branches, the file references get messed up, causing the NPE. I tried a few other implementations, one where we generate the html config files first, but it get pretty complicated. I had considered cleanup, but then thought that we don't cleanup /repositories or /reports, so why worry? I can certainly add cleanup to this.  I could create a "configHtml" directory that deleted after report generation.

Reply:
We are usually running diff.groovy from ./ci-temp/ in the Checkstyle project, so file refs get messed up when we switch the branch from master to patch (or vice-versa).

Reply:
Is the issue just when you leave the config files in the same repo as the checkstyle repo that is being regressed on?

Reply:
Yes, https://github.com/checkstyle/checkstyle/pull/9186/files#r560292164 is a good example.  I think this is ultimately the same problem in regression-tool, but I didn't dig into that repo too deep since it doesn't seem to be used anymore.

 My other thought to overcome this was to just create an option so that we can skip the final report generation, since when `diff.groovy` is used in this way, we don't need a final report anyway.

Reply:
That explains the issue. Truthfully it seems it should be given that you can't leave files in the repo being regressed on as the util will switch branches.

If @romani is fine with this then ok.